docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,419 @@
+
+When using variables, you declare the variable, specify a type, and assign
+values to the variable. The following example declares a variable `i` of type
+`int`, and assigns it a value of `10`:
+
+[source,Painless]
 ----
-{
-"acknowledged": true
-}
+int i;  <1>
+i = 10; <2>
 ----
-////
-=====
+<1> Declare `int i` and store default `int 0` to `i`
+<2> Store `int 10` to `i`
 
-We recommend testing and benchmarking any indexing changes before deploying them
-in production.
+When declaring a variable, specify the type followed by an identifier. After
+declaring variables, you can use them in scripts by referring to their
+identifier.
+
+Use the assignment operator `'='` to store a value in a variable for use in
+subsequent operations. Any operation that produces a value can be assigned to
+any variable as long as the types are the same, or the resultant type can be
+implicitly cast to the variable type.
+
+[[script-spec-operators]]
+=== Operators
+An operator is the most basic action that can be taken to evaluate values in a
+script. An expression is one-to-many consecutive operations. Precedence is the
+order in which an operator will be evaluated relative to another operator.
+Associativity is the direction within an expression in which a specific
+operator is evaluated.
+
+[[script-spec-statements]]
+=== Statements
 
-[discrete]
-[[modules-scripting-errors]]
-=== Script errors
-Elasticsearch returns error details when there is a compliation or runtime
-exception.  The contents of this response are useful for tracking down the
-problem.
 
-experimental[]
+[[script-spec-functions]]
+=== Functions
+
+
+[[modules-scripting-using-caching]]
+=== Modify script caching
+
+All scripts are cached by default so that they only need to be recompiled
+when updates occur. By default, scripts do not have a time-based expiration, but
+you can change this behavior by using the `script.cache.expire` setting.
+You can configure the size of this cache by using the `script.cache.max_size` setting.
+For most contexts, the default cache size is `100`. For ingest contexts, the
+default cache size is `200`.
 
-The contents of `position` are experimental and subject to change.
+NOTE: The size of scripts is limited to 65,535 bytes. This can be
+changed by setting `script.max_size_in_bytes` setting to increase that soft
+limit, but if scripts are really large then a
+<<modules-scripting-engine,native script engine>> should be considered.
+
+[[scripts-and-search-speed]]
+=== Scripts and search speed
+
+Scripts can't make use of {es}'s index structures or related optimizations. This
+can sometimes result in slower search speeds.
+
+If you often use scripts to transform indexed data, you can speed up search by
+making these changes during ingest instead. However, that often means slower
+index speeds.
+
+[[modules-scripting-stored-scripts]]
+=== Stored scripts
+
+Scripts may be stored in and retrieved from the cluster state using the
+`_scripts` end-point.
```

Comment:
Should this be part of the using section? This seems closer to general scripting info to me.

Reply:
These sections are just a placeholder right now that still need to be worked in.